Navigation/include/ActsLUXEPipeline/LUXEMeasurementsCreator.hpp

```diff
@@ -0,0 +1,246 @@
+                return;
+            }
+
+            // only generate measurements on surfaces
+            if (!navigator.currentSurface(state.navigation)) {
+//                result.fullTrack.push_back(stepper.position(state.stepping));
+                return;
+            }
+
+            const Acts::Surface &surface = *navigator.currentSurface(state.navigation);
+            const Acts::GeometryIdentifier geoId = surface.geometryId();
+            // only generate measurements on sensitive surface
+            if (!geoId.sensitive()) {
+                ACTS_VERBOSE("Create no measurements on non-sensitive surface " << geoId);
+                return;
+            }
+
+            // Apply global to local
+            Acts::Vector3 pos3 = stepper.position(state.stepping);
+            Acts::Vector2 loc =
+                    surface.globalToLocal(state.geoContext, pos3,
+                                          stepper.direction(state.stepping)).value();
+            // The truth info
+            Acts::BoundVector parameters = Acts::BoundVector::Zero();
+            parameters[Acts::eBoundLoc0] = loc[Acts::eBoundLoc0];
+            parameters[Acts::eBoundLoc1] = loc[Acts::eBoundLoc1];
+            const auto &direction = stepper.position(state.stepping);
+            parameters[Acts::eBoundPhi] = Acts::VectorHelpers::phi(direction);
+            parameters[Acts::eBoundTheta] = Acts::VectorHelpers::theta(direction);
+            parameters[Acts::eBoundQOverP] = state.stepping.pars[Acts::eFreeQOverP];
+            parameters[Acts::eBoundTime] = state.stepping.pars[Acts::eFreeTime];
+            // Construct a particle object
+            Acts::Vector4 particlePos = {pos3[0],pos3[1],pos3[2],parameters[Acts::eBoundTime]};
+            ActsFatras::Particle tempParticle =
+                    makeParticle(
+                            stepper.particleHypothesis(state.stepping).absolutePdg(),stepper.direction(state.stepping),
+                            stepper.charge(state.stepping)/parameters[Acts::eBoundQOverP],
+                            particlePos);
+            std::vector<Acts::MaterialSlab> layerMaterials = {makeSiliconSlab()};
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            auto scattering = ActsFatras::GaussianMixtureScattering();
+            ActsFatras::BetheBloch BBProcess;
+            ActsFatras::BetheHeitler BHProcess;
+
+            Acts::BoundVector scatteredParameters = parameters;
+            ActsFatras::Particle beforeParticle = tempParticle;
+            for (auto material:layerMaterials) {
+                float p1 = tempParticle.absoluteMomentum();
+                auto scatter = scattering(gen, material, tempParticle);
+                auto BBLoss = BBProcess(gen, material, tempParticle);
+                auto BHLoss = BHProcess(gen, material, tempParticle);
+                float p2 = tempParticle.absoluteMomentum();
+                if (p2 < p1/2) {
```

Comment:
What is going on here? Just curious

Reply:
just ad-hoc correction, 1 out of every ~100000 the energy after passing through material was almost 0 and it broke the navigator